common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MSSTSBrokerAuthoriztionRequest.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.providers.microsoft.microsoftsts;
+
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ClientException;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+public class MSSTSBrokerAuthoriztionRequest extends MicrosoftStsAuthorizationRequest {

```

Comment:
This class contains significant duplication of method names (and some constants) with `AADBrokerAuthorizationRequest` -- consider adding an `IBrokerAuthorizationRequest` interface and `BrokerAuthorizationRequest` superclass (IMO, both might be useful) -- the `BrokerAuthorizationRequest` class would very likely be an `abstract` implementation.

I would also consider creating a `MicrosoftAuthorizationRequest` superclass to share non-broker authorization code that is Microsoft-specific but shareable between `AAD` and `MSSTS`

Consider using [composition ](https://www.artima.com/designtechniques/compoinh.html)to solve [the diamond problem](https://en.wikipedia.org/wiki/Multiple_inheritance#The_diamond_problem) created by having `BrokerAuthorizationRequest` and `MicrosoftAuthorizationRequest` superclasses